packages/react-sdk/src/Provider.tsx

```diff
@@ -36,32 +43,83 @@ export class OptimizelyProvider extends React.Component<
   OptimizelyProviderProps,
   OptimizelyProviderState
 > {
-  sdkWrapper: UserWrappedOptimizelySDK
-
   constructor(props: OptimizelyProviderProps) {
     super(props)
+    const { optimizely, userId, userAttributes, user} = props
+
+    // check if user id/attributes are provided as props and set them ReactSDKClient
+    let finalUser: {
+      id: string
+      attributes: optimizely.UserAttributes
+    } | null = null
+
+    if (user) {
+      if ('then' in user) {
+        user.then(user => {
+          optimizely.setUser(user)
+        })
+      } else {
+        finalUser = {
+          id: user.id,
+          attributes: user.attributes || {},
+        }
+      }
+    } else if (userId) {
+      finalUser = {
+        id: userId,
+        attributes: userAttributes || {},
+      }
+      // deprecation warning
+      console.warn(
```

Comment:
Should this be logged with the logger instead of polluting the console stream? In other SDKs we log to the logger when we deprecate stuff.